### DISCUSSION TITLE
Wingman App

### DISCUSSION THREAD

**Discussion by Winehorn** [Wollivan/SquadronLeader#33] [Ideas]
General space to talk about possible features for a companion and technologies we want to use.

# Features
Core feature should probably be stat tracking as this is something the tin doesn't contain at the moment.

Because I love data it would be awesome if we allow users to upload the results of their dogfights so we could visualize which stats prove to be the best! (And maybe add some balancing rules based on this)

Another feature could be an AI system for co-op or solo plays.

# Technologies
I have a bit of experience with Flutter for cross platform development. I read that @Wollivan works with React, which is something I don't know but our FE devs at work use it so I'd be open to learn it :)

**Comment by Wollivan** [comment]
This is such a great idea!!!

I'm totally open to flutter being used, I also know @mevdev is an app developer so he might want to weigh in?

Great idea about collecting data for game balancing, that's also something we could start now. If anyone can think of an elegant way to collect and display that data I'm all ears!

For the AI, also great! I want to make sure the AI can also be fully analogue for people without phones, and maybe even a printable tool to assist, but there's no reason we can't have 2 versions.

Really great post idea, thank you!

**Comment by penyuan** [comment]
Really excited by a possible app! 🚀 

1. I agree with @Wollivan that the AI should be fully playable in analogue *without* the app. This is to make the game accessible to those without a device for the app, but is also important as a play style thing.
2. In the spirit of this game, I request that the app be fully open source as well! 😃

**Reply by Winehorn** [reply]
Agree with both points!

**Comment by Wollivan** [comment]
> In the spirit of this game, I request that the app be fully open source as well! 😃

Strong agree!

**Comment by mevdev** [comment]
This sounds like a great idea. With a game the limitation of the size of the table that could make actually creating the whole game pretty straightforward.

Native all the way, as well. Making a 'companion' app would be rather simple. 

What would be desired for it?
- Hit points for planes
- dice rolling
- Altitude info
- Rules summary

**Comment by Wollivan** [comment]
Yeah that list is perfect, and also something like a game mode, where you have all the players stats on display, and color coding etc

**Comment by Wollivan** [comment]
https://squadronleader.wollivan.dev/tracker

@Winehorn @penyuan @mevdev 

Basic stat tracker on the site.

It doesn't do all the things we discussed, but it's a start!
